### PR TITLE
build: If on a pre-release, use the SemVer with SHA1 metadata

### PR DESCRIPTION
**Pull request by sschuberth**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -43,7 +43,7 @@ semver {
 
 // Only override a default version (which usually is "unspecified"), but not a custom version.
 if (version == Project.DEFAULT_VERSION) {
-    version = semver.version
+    version = semver.semVersion.takeIf { it.isPreRelease } ?: semver.version
```
Comment (fviernau) [inline build.gradle.kts:46]: Does `it.isPreRelease` return true for any `semver` value which is not a "release version" ?

Comment (sschuberth) [reply build.gradle.kts:46]: I'm not sure what you definition of a "release version" is, so I'll answer differently: `it.isPreRelease` returns `true` for any SemVer that contains the `-` separated part of the version (not the `+` separated part). You might want to have a look [here](https://github.com/jmongard/Git.SemVersioning.Gradle/blob/ac9b08ef51460d5b3cab126856cf0923676fe76e/src/test/kotlin/git/semver/plugin/semver/SemVersionTest.kt#L22-L53) to get the idea.

Comment (fviernau) [reply build.gradle.kts:46]: Ok, so let me re-phrase the question: Does the new code guarantee that for any version which is not tagged with a (release) Git tag, the version string contains the SHA1?

Comment (sschuberth) [reply build.gradle.kts:46]: Yes.

**Review by fviernau — APPROVED** [review]

**Review by fviernau — APPROVED** [review]

**Pull request merged**